ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js

```diff
@@ -0,0 +1,94 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import {
+  BannerAlert,
+  ButtonLink,
+  Icon,
+  IconName,
+  IconSize,
+  Text,
+} from '../../component-library';
+import {
+  AlignItems,
+  Color,
+  Severity,
+  Size,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { I18nContext } from '../../../../.storybook/i18n';
+
+import Disclosure from '../../ui/disclosure';
+import { DisclosureVariant } from '../../ui/disclosure/disclosure.constants';
+import {
+  SecurityProvider,
+  SECURITY_PROVIDER_CONFIG,
+} from '../../../../shared/constants/security-provider';
+
+function SecurityProviderBannerAlert({
+  description,
+  details,
+  provider,
+  severity,
+  title,
+}) {
+  const t = useContext(I18nContext);
+
+  return (
+    <BannerAlert
+      title={title}
+      severity={severity}
+      marginTop={4}
+      marginRight={4}
+      marginLeft={4}
+    >
+      <Text variant={TextVariant.bodySm} as="h6" marginTop={2}>
+        {description}
+      </Text>
```

Comment:
Could you use the `description` prop inside the `BannerAlert`? I think it should be `bodyMd`? Our typography is responsive so if you see 14px the design might be using the small screen typography. It should be 16px on larger screens. Also I know a lot of the Text are rendering as `h6` this is probably not the best semantic mark up but I think was a decision instead of getting new contributors to update e2e tests. Probably not the best in hindsight. I think it could just be a `p` tag which is what bodySm renders by default

Reply:
I see now. `description` is inherited from `BannerBase`. Got it 

I was carrying over code and preserving styles from our OpenSea SecurityProviderMessage component. These are great updates 👍  Thank you!

applied in [647eda9](https://github.com/MetaMask/metamask-extension/pull/19825/commits/647eda970ad7c07a4c4a032db766d2aab10cfc73) and [a7fdeee](https://github.com/MetaMask/metamask-extension/pull/19825/commits/a7fdeee4190606a17d9a6d1bf685cdecc7ededc5)

I also adjusted the Disclosure component text size here https://github.com/MetaMask/metamask-extension/pull/19770/commits/8ac92488697a27152e00e9c50d1fe0d23ff95573

One thing using the `description` param is that is a lack of padding between the title and description which does not match the figma design. Any suggestions? @georgewrmarshall

Reply:
before: 
<img width="794" alt="Screen Shot 2023-07-10 at 9 15 35 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/57401982-8d2a-4621-ab1f-1afa7dc4b2cf">

after using `description` param:
<img width="1046" alt="Screen Shot 2023-07-12 at 12 08 47 AM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/3c6008cf-8311-4776-a4f4-e58530919cf0">

Reply:
Hey @georgewrmarshall! hope you had a nice weekend. In addition to the lack of padding/margin between the title and the description, I realized there is another issue we have using the description BannerAlert/BannerBase property instead of passing children elements. 

The description currently expects a string property. The problem is that SecurityProviderBannerAlert will allow hyperlinks, or other HTML elements, in the description. 

error: 
> BannerBase   > checkPropTypes.js:20 Warning: Failed prop type: Invalid prop `description` of type `object` supplied to `BannerAlert`

  Do you have suggestions to address the two issues above?

Reply:
In the meantime I have reverted the use of description param to fix the console and UI issues. I think it would be better to address this in a separate PR to unblock dependent work ahead 

Revert commit https://github.com/MetaMask/metamask-extension/pull/19825/commits/1c3d5508f00ad8f6f8e23ee42cd19e1c9aab2fda

Reply:
I am thinking, maybe, we can do something similar to our plans with the Disclosure component and add support using Text as=“div”